storage/client_status_test.go

```diff
@@ -0,0 +1,221 @@
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Bram Gruneir (bram.gruneir@gmail.com)
+
+package storage_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/gossip"
+	"github.com/cockroachdb/cockroach/kv"
+	"github.com/cockroachdb/cockroach/multiraft"
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/rpc"
+	"github.com/cockroachdb/cockroach/storage"
+	"github.com/cockroachdb/cockroach/storage/engine"
+	"github.com/cockroachdb/cockroach/util"
+	"github.com/cockroachdb/cockroach/util/hlc"
+	"github.com/cockroachdb/cockroach/util/leaktest"
+	gogoproto "github.com/gogo/protobuf/proto"
+)
+
+// createTestStoreWithScanner mimics client_test's createTestStore but
+// specifically sets the range scanner interval to 10ms.
+func createTestStoreWithScanner(t *testing.T) (*storage.Store, *util.Stopper) {
+	eng := engine.NewInMem(proto.Attributes{}, 10<<20)
+	clock := hlc.NewClock(hlc.NewManualClock(0).UnixNano)
+	stopper := util.NewStopper()
+	rpcContext := rpc.NewContext(hlc.NewClock(hlc.UnixNano), rpc.LoadInsecureTLSConfig(), stopper)
+	ctx := storage.TestStoreContext
+	ctx.Gossip = gossip.New(rpcContext, gossip.TestInterval, gossip.TestBootstrap)
+	lSender := kv.NewLocalSender()
+	sender := kv.NewTxnCoordSender(lSender, clock, false, stopper)
+	ctx.Clock = clock
+	ctx.DB = client.NewKV(nil, sender)
+	ctx.DB.User = storage.UserRoot
+	ctx.Transport = multiraft.NewLocalRPCTransport()
+	ctx.ScanInterval = 10 * time.Millisecond
+	// TODO(bdarnell): arrange to have the transport closed.
+	store := storage.NewStore(ctx, eng)
+	if err := store.Bootstrap(proto.StoreIdent{NodeID: 1, StoreID: 1}, stopper); err != nil {
+		t.Fatal(err)
+	}
+	lSender.AddStore(store)
+	if err := store.BootstrapRange(); err != nil {
+		t.Fatal(err)
+	}
+	if err := store.Start(stopper); err != nil {
+		t.Fatal(err)
+	}
+	return store, stopper
+}
+
+// compareStoreStatus ensures that the actual store status for the passed in
+// store is updated correctly. It checks that the StoreID, NodeID and
+// RangeCount are exactly correct and that the bytes and counts for Live, Key
+// and Val are at least the expected value.  The latest actual stats are
+// returned.
+func compareStoreStatus(t *testing.T, store *storage.Store, expectedStoreStatus *proto.StoreStatus) *proto.StoreStatus {
+	storeStatusKey := engine.StoreStatusKey(int32(store.Ident.StoreID))
+	gArgs, gReply := getArgs(storeStatusKey, 1, store.Ident.StoreID)
+	if err := store.ExecuteCmd(gArgs, gReply); err != nil {
+		t.Fatalf("failure getting store status: %s", err)
+	}
+	if gReply.Value == nil {
+		t.Errorf("could not find store status at: %s", storeStatusKey)
+	}
+	storeStatus := &proto.StoreStatus{}
+	if err := gogoproto.Unmarshal(gReply.Value.GetBytes(), storeStatus); err != nil {
+		t.Errorf("could not unmarshal store status: %+v", gReply)
```

Comment:
This should probably be a Fatalf instead of Errorf, since it is meaningless to continue past an unmarshaling error.